src/Http/Http.Results/src/ResultsCache.StatusCodes.cs

```diff
@@ -0,0 +1,153 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// <auto-generated />
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Http;
+
+using System.CodeDom.Compiler;
+
+[GeneratedCode("TextTemplatingFileGenerator", "")]
+internal partial class ResultsCache
+{
+    private static StatusCodeHttpResult? _status100Continue;
```

Comment:
Should these be `static readonly`s
```suggestion
    private static readonly StatusCodeHttpResult? s_status100Continue = new(StatusCodes.Status100Continue);
```
And in the switch below
```c#
        return statusCode switch
        {
            StatusCodes.Status100Continue => s_status100Continue,
            ...
```
That way the lazy-initialization is shifted to the runtime, and by `beforefieldinit` it's done on first access.

With tiered-compilation hot `StatusCodeHttpResult`s could already be initialized, so no extra (runtime-) check is needed anymore.
Otherwise it's pretty the same if there's a `?? new(StatusCodes.XYZ)` or if that check is done by the runtime (I guess).

---

Of course the change in the `tt`-file below.

Reply:
@gfoidl I really appreciate your comment. I did not try the tiered-compilation thing yet however the `beforefieldinit` indeed do a lazy-initialization but based on my tests when the initialization happens all the ~ 60 fields are initialized and all memory is allocated. The performance improvement is minimum compared with the null check approach.

Initial allocation cost (First time access)

Method | Gen 0 | Gen 1 | Allocated
-- | -- | -- | --
StaticCacheWithSwitchExpression | 0.0095 | - | 2,000 B
DynamicCacheWithSwitchExpression | 0.0001 | - | 24 B

<!--EndFragment-->
</body>
</html>

Method | Mean | Error | StdDev | Gen 0 | Allocated
-- | -- | -- | -- | -- | --
StaticCacheWithSwitchExpression | 1.867 ns | 0.0045 ns | 0.0042 ns | - | -
DynamicCacheWithSwitchExpression | 1.889 ns | 0.0143 ns | 0.0119 ns | - | -

Reply:
Drop _status100Continue, that's a response code that's only ever returned by the server, not the app.